### ISSUE TITLE
Illegal type annotation on a type variable that's been reassigned

### ISSUE THREAD

**Issue by RileyEv** [microsoft/pyright#3114]
**Describe the bug**
I get an illegal type annoration error on a type variable that has been assigned to a new variable. For error details see pyright output below.

**To Reproduce**
See code below

**Expected behavior**
In the code shown, the use of `A` in class `B` is found to be correctly typed. I would expect the same to be the case for `A_` in class `B_`, since `A = A_`.

**Screenshots or Code**

`a.py`
```python
from typing import TypeVar


A = TypeVar("A")
```

`b.py`
```python
from typing import Generic

from a import A

A_ = A


class B(Generic[A]):
    pass


class B_(Generic[A_]):
    pass
```

Pyright output
```
$ pyright
npx: installed 1 in 2.995s
Loading configuration file at /home/riley/Projects/testtype/pyrightconfig.json
Assuming Python version 3.10
Assuming Python platform Linux
No include entries specified; assuming /home/riley/Projects/testtype
Auto-excluding **/node_modules
Auto-excluding **/__pycache__
Auto-excluding **/.*
stubPath /home/riley/Projects/testtype/typings is not a valid directory.
Searching for source files
Found 2 source files
/home/riley/Projects/testtype/b.py
  /home/riley/Projects/testtype/b.py:12:18 - error: Illegal type annotation: variable not allowed unless it is a type alias (reportGeneralTypeIssues)
  /home/riley/Projects/testtype/b.py:12:18 - error: Type argument for "Generic" must be a type variable
2 errors, 0 warnings, 0 informations 
Completed in 0.583sec
```

**VS Code extension or command-line**
Issue appears via cli and emacs lsp mode.

**Comment by erictraut** [comment]
I think the behavior is correct.

TypeVars should be named the same as the name they were given when created. For example, if you create `A = TypeVar("A")`, you should not reassign it to a variable called `A_`.

Also, type annotation expressions are not allowed to use variables.
